rosbag2_py/src/rosbag2_py/_info.cpp

```diff
@@ -77,15 +82,62 @@ class Info
     const std::string & sorting_method)
   {
     std::vector<std::shared_ptr<rosbag2_cpp::rosbag2_service_info_t>> all_services_info;
+    std::vector<std::shared_ptr<rosbag2_cpp::rosbag2_action_info_t>> all_actions_info;
+
     for (auto & file_info : metadata_info.files) {
-      auto services_info = info_->read_service_info(
+      std::vector<std::shared_ptr<rosbag2_cpp::rosbag2_service_info_t>> output_service_info;
+      std::vector<std::shared_ptr<rosbag2_cpp::rosbag2_action_info_t>> output_action_info;
+      info_->read_service_action_info(
+        output_service_info,
+        output_action_info,
         uri + "/" + file_info.path,
         metadata_info.storage_identifier);
-      if (!services_info.empty()) {
+      if (!output_service_info.empty()) {
         all_services_info.insert(
           all_services_info.end(),
-          services_info.begin(),
-          services_info.end());
+          output_service_info.begin(),
+          output_service_info.end());
+      }
+      if (!output_action_info.empty()) {
+        all_actions_info.insert(
+          all_actions_info.end(),
+          output_action_info.begin(),
+          output_action_info.end());
+      }
+    }
+
+    // Fill in the number of feedback and status messages for the action info from
+    // metadata_info.topics_with_message_count
+    if (!all_actions_info.empty()) {
+      for (auto & [topic_metadata, message_count] : metadata_info.topics_with_message_count) {
+        if (rosbag2_cpp::is_topic_related_to_action(topic_metadata.name, topic_metadata.type)) {
+          auto action_interface_type =
+            rosbag2_cpp::get_action_topic_type_from_topic_name(topic_metadata.name);
```

Comment:
The logic and runtime complexity could be simplified with the following
```suggestion
        auto action_interface_type = rosbag2_cpp::get_action_interface_type(topic_metadata.name);
        if (rosbag2_cpp::is_topic_belong_to_action(action_interface_type, topic_metadata.type)) {
```